crates/project_panel/src/project_panel.rs

```diff
@@ -1178,11 +1178,25 @@ impl ProjectPanel {
                             let a_is_file = components_a.peek().is_none() && entry_a.is_file();
                             let b_is_file = components_b.peek().is_none() && entry_b.is_file();
                             let ordering = a_is_file.cmp(&b_is_file).then_with(|| {
-                                let name_a =
-                                    UniCase::new(component_a.as_os_str().to_string_lossy());
-                                let name_b =
-                                    UniCase::new(component_b.as_os_str().to_string_lossy());
-                                name_a.cmp(&name_b)
+                                let a_as_number = Path::new(component_a.as_os_str())
+                                    .file_stem()
+                                    .and_then(|s| s.to_str())
+                                    .map(|stem| stem.parse::<i32>());
+                                let b_as_number = Path::new(component_b.as_os_str())
+                                    .file_stem()
+                                    .and_then(|s| s.to_str())
+                                    .map(|stem| stem.parse::<i32>());
```

Comment:
Is it true that right now this only works for file stems consisting solely of numbers?

I wonder if we should try to detect if the file stem starts with a sequence of digits and then sort based on the numeric representation of that first, and then sort by the remaining string representation.

I’m thinking of files like migration names that could start with a numeric timestamp and then have a `-migration-name` suffix.

Reply:
You're right, that's a good point. Let me take a stab at that.

Reply:
Implemented together in b4386cab2